autopilot/autopilot.go

```diff
@@ -603,23 +597,27 @@ func (ap *Autopilot) performWalletMaintenance(ctx context.Context) error {
 	for _, txn := range pending {
 		for _, mTxnID := range ap.maintenanceTxnIDs {
 			if mTxnID == types.TransactionID(txn.ID) {
-				l.Debugf("wallet maintenance skipped, pending transaction found with id %v", mTxnID)
+				l.Infof("wallet maintenance skipped, pending transaction found with id %v", mTxnID)
 				return nil
 			}
 		}
 	}
 
-	// figure out the amount per output
+	// check whether the wallet needs to be redistributed
 	wantedNumOutputs := 10
-	amount := contractor.InitialContractFunding.Div64(uint64(wantedNumOutputs))
+	amount := contractor.InitialContractFunding.Mul64(10)
+	if balance.Cmp(amount.Mul64(uint64(wantedNumOutputs))) < 0 {
+		l.Warnf("wallet maintenance skipped, wallet balance %v is too low to redistribute into meaningful outputs", balance)
+		return nil
+	}
 
 	// redistribute outputs
-	ids, err := b.WalletRedistribute(ctx, wantedNumOutputs, amount)
+	ids, err := b.WalletRedistribute(ctx, wantedNumOutputs, balance.Div64(11))
```

Comment:
```suggestion
	ids, err := b.WalletRedistribute(ctx, wantedNumOutputs, amount)
```
Otherwise the amount you computed above is only used for the check and logging

Reply:
I did this on purpose but I'm fine to commit the suggestion. Dividing it by 11 avoids an edge case where you have exactly enough to redistribute but not enough to cover the txn fee.